app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,308 @@
+        },
+        footer: {
+            height: 40,
+            width: '100%',
+        },
+    };
+});
+
+const EditProfile = ({
+    closeButtonId,
+    componentId,
+    currentUser,
+    isModal,
+    isTablet,
+    lockedFirstName,
+    lockedLastName,
+    lockedNickname,
+    lockedPosition,
+}: EditProfileProps) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+
+    const [userInfo, setUserInfo] = useState<UserInfo>({
+        email: currentUser.email,
+        firstName: currentUser.firstName,
+        lastName: currentUser.lastName,
+        nickname: currentUser.nickname,
+        position: currentUser.position,
+        username: currentUser.username,
+    });
+    const [canSave, setCanSave] = useState(false);
+    const [error, setError] = useState<string | undefined>();
+
+    const [profileImage] = useState<undefined>();
+
+    const [updating, setUpdating] = useState(false);
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'update-profile':
+                        submitUser();
+                        break;
+                    case closeButtonId:
+                        close();
+                        break;
+                }
+            },
+        }, componentId);
+
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+
+        return () => {
+            unsubscribe.remove();
+            backHandler.remove();
+        };
+    }, [userInfo, isTablet, profileImage]);
+
+    useEffect(() => {
+        if (!isTablet) {
+            setButtons(componentId, {rightButtons: [rightButton]});
+        }
+    }, [isTablet]);
+
+    const style = getStyleSheet(theme);
+    const service = currentUser.authService;
+
+    const rightButton = {
+        id: 'update-profile',
+        enabled: false,
+        showAsAction: 'always' as const,
+        testID: 'edit_profile.save.button',
+        color: theme.sidebarHeaderTextColor,
+        text: intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'}),
+    };
+
+    const close = () => {
+        if (isModal) {
+            dismissModal({componentId});
+        } else if (isTablet) {
+            DeviceEventEmitter.emit(Events.ACCOUNT_SELECT_TABLET_VIEW, '');
+        } else {
+            popTopScreen(componentId);
+        }
+
+        return true;
+    };
+
+    const enableSaveButton = (value: boolean) => {
+        if (isTablet) {
+            setCanSave(value);
+        } else {
+            const buttons = {
+                rightButtons: [{...rightButton, enabled: value}],
+            };
+
+            setButtons(componentId, buttons);
+        }
+    };
+
+    const submitUser = async () => {
```

Comment:
you should apply a `preventDoubleTap` here

Reply:
Need my help?

Reply:
Done